shell/platform/windows/client_wrapper/flutter_window_controller.cc

```diff
@@ -0,0 +1,407 @@
+            auto const& that{it2->second};
+            lock.unlock();
+            DestroyWindow(that->GetHandle());
+            lock.lock();
+            it2 = windows_.begin();
+          }
+        }
+
+        SendOnWindowDestroyed(view_id);
+      }
+    }
+      return 0;
+    case WM_SIZE: {
+      std::lock_guard lock{mutex_};
+      auto const it{std::find_if(windows_.begin(), windows_.end(),
+                                 [hwnd](auto const& window) {
+                                   return window.second->GetHandle() == hwnd;
+                                 })};
+      if (it != windows_.end()) {
+        auto const view_id{it->first};
+        SendOnWindowChanged(view_id);
+      }
+    } break;
+    default:
+      break;
+  }
+
+  if (auto* const window{Win32Window::GetThisFromHandle(hwnd)}) {
+    return window->MessageHandler(hwnd, message, wparam, lparam);
+  }
+  return DefWindowProc(hwnd, message, wparam, lparam);
+}
+
+void FlutterWindowController::SendOnWindowCreated(
+    FlutterViewId view_id,
+    std::optional<FlutterViewId> parent_view_id) const {
+  if (channel_) {
+    channel_->InvokeMethod(
+        "onWindowCreated",
+        std::make_unique<EncodableValue>(EncodableMap{
+            {EncodableValue("viewId"), EncodableValue(view_id)},
+            {EncodableValue("parentViewId"),
+             parent_view_id ? EncodableValue(parent_view_id.value())
+                            : EncodableValue()}}));
+  }
+}
+
+void FlutterWindowController::SendOnWindowDestroyed(
+    FlutterViewId view_id) const {
+  if (channel_) {
+    channel_->InvokeMethod(
+        "onWindowDestroyed",
+        std::make_unique<EncodableValue>(EncodableMap{
+            {EncodableValue("viewId"), EncodableValue(view_id)},
+        }));
+  }
+}
+
+void FlutterWindowController::SendOnWindowChanged(FlutterViewId view_id) const {
+  if (channel_) {
+    auto const size{GetWindowSize(view_id)};
+    channel_->InvokeMethod(
+        "onWindowChanged",
+        std::make_unique<EncodableValue>(EncodableMap{
+            {EncodableValue("viewId"), EncodableValue(view_id)},
+            {EncodableValue("size"),
+             EncodableValue(EncodableList{EncodableValue(size.width),
+                                          EncodableValue(size.height)})},
+            {EncodableValue("relativePosition"), EncodableValue()},  // TODO
+            {EncodableValue("isMoving"), EncodableValue()}}));       // TODO
+  }
+}
+
+void FlutterWindowController::HandleCreateWindow(WindowArchetype archetype,
+                                                 MethodCall<> const& call,
+                                                 MethodResult<>& result) {
+  auto const* const arguments{call.arguments()};
+  auto const* const map{std::get_if<EncodableMap>(arguments)};
+  if (!map) {
+    result.Error(kErrorCodeInvalidValue, "Method call argument is not a map.");
+    return;
+  }
+
+  std::wstring const title{ArchetypeToWideString(archetype)};
```

Comment:
Setting the title seems to the archetype seems like debugging code that shouldn't be here anymore?

Reply:
Yes, it's just a placeholder. Setting the title is straightforward. I'll do it.